tests/gfmdataprocessor/headers.js

```diff
@@ -0,0 +1,111 @@
+/**
+ * @license Copyright (c) 2003-2016, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+import { testDataProcessor as test } from '/tests/markdown-gfm/_utils/utils.js';
+
+describe( 'GFMDataProcessor', () => {
+	describe( 'headers', () => {
+		it( 'should process level 1 header #1', () => {
+			test(
+				'# Level 1',
+
+				// GitHub is rendering as:
+				// <h1>Level 1</h1>
+				'<h1 id="level-1">Level 1</h1>'
```

Comment:
> When you comment that GH is rendering as something and below we have something else, do you mean that simply the results differ? That the GFMDP will produce a different result?

I was checking each test how it is rendered by GitHub and reported it in comments if there wasn't an easy way to fix this. I will report all these issues in separate follow up as you suggested. (BTW. This header id can be easily removed in `marked` renderer - I will do this for now, we can always add it back later. )